libpod/runtime_ctr.go

```diff
@@ -238,41 +238,61 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force bool)
 		return errors.Wrapf(ErrCtrExists, "container %s has dependent containers which must be removed before it: %s", c.ID(), depsStr)
 	}
 
-	// Clean up network namespace, cgroups, mounts
-	if err := c.cleanup(); err != nil {
-		return err
-	}
-
-	// Stop the container's storage
-	if err := c.teardownStorage(); err != nil {
-		return err
-	}
-
+	var cleanup_err error
 	// Remove the container from the state
 	if c.config.Pod != "" {
 		if err := r.state.RemoveContainerFromPod(pod, c); err != nil {
-			return err
+			if cleanup_err == nil {
+				cleanup_err = err
+			} else {
+				cleanup_err = errors.Wrapf(err, "%v", cleanup_err)
```

Comment:
Is it better to log the error instead of building a chain of errors with wrapf?

Reply:
> Is it better to log the error instead of building a chain of errors with wrapf?

How about a pattern like:

```
if err2 := r.state.RemoveContainerFromPod(pod, c); err2 != nil {
    logrus.Warnf("removing container %s: %v", c.id, err2)
    if err == nil {
      err = err2
    }
}
```

so all of these errors get logged as non-fatal warnings, and the first of them gets returned to callers (who can decide if the presence of an error is fatal or not).  I think that would work fairly well, unless the caller needed to condition fatal-ness on the particular error, in which case it would need to know about everything that failed, and not just the first failure.

Reply:
Would you stuff the latest error into cleanup_err and return that still?  I think I'd prefer logging each then returning the latest with some kind of added "There were one or more errors cleaning then environment" type of error.  I think I'd lean with that approach.